pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -0,0 +1,259 @@
+	return secretBundle, nil
+}
+
+// implement store.Client.GetSecretMap Interface.
+// retrieve ALL secrets in a specific keyvault with the name ref.Name.
+func (a *Azure) GetSecretMap(ctx context.Context, ref esv1alpha1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
+	secretMap, err := a.iAzure.getKeyVaultSecrets(ctx, ref.Key, ref.Version, "", true)
+	return secretMap, err
+}
+
+// getCertBundle returns the certificate bundle.
+func getCertBundleForPKCS(certificateRawVal string, certBundleOnly, certKeyOnly bool) (bundle string, err error) {
+	pfx, err := base64.StdEncoding.DecodeString(certificateRawVal)
+
+	if err != nil {
+		return bundle, err
+	}
+	blocks, _ := pkcs12.ToPEM(pfx, "")
+
+	for _, block := range blocks {
+		// skip the private key if looking for the cert only
+		if block.Type == "PRIVATE KEY" && certBundleOnly {
+			continue
+		}
+		// no headers
+		if block.Type == "PRIVATE KEY" {
+			pkey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+			if err != nil {
+				panic(err)
+			}
+			derStream := x509.MarshalPKCS1PrivateKey(pkey)
+			block = &pem.Block{
+				Type:  "RSA PRIVATE KEY",
+				Bytes: derStream,
+			}
+			if certKeyOnly {
+				bundle = string(pem.EncodeToMemory(block))
+				break
+			}
+		}
+
+		block.Headers = nil
+		bundle += string(pem.EncodeToMemory(block))
+	}
+	return bundle, nil
+}
+
+// consolidated method to retrieve secret value or secrets list based on whether or not a secret name is passed.
+// if the secret is of type PKCS then this is a cerificate that needs some decoding.
+func (a *Azure) getKeyVaultSecrets(ctx context.Context, vaultName, version, secretName string, withTags bool) (map[string][]byte, error) {
+	basicClient := a.baseClient
+	secretsMap := make(map[string][]byte)
+	certBundleOnly := false
+	certKeyOnly := false
+	secretNameinBE := secretName
+
+	if secretName != "" {
+		nameSplitted := strings.Split(secretName, "_")
+		if nameSplitted[len(nameSplitted)-1] == "CRT" {
+			secretNameinBE = nameSplitted[0]
+			certBundleOnly = true
+		}
+		if nameSplitted[len(nameSplitted)-1] == "KEY" {
+			secretNameinBE = nameSplitted[0]
+			certKeyOnly = true
+		}
+
+		secretResp, err := basicClient.GetSecret(context.Background(), "https://"+vaultName+".vault.azure.net", secretNameinBE, version)
+		if err != nil {
+			return nil, err
+		}
+		secretValue := *secretResp.Value
+
+		// Azure currently supports only PKCS#12 or PEM, PEM will be taken as it is, PKCS needs processing
+		if secretResp.ContentType != nil && *secretResp.ContentType == "application/x-pkcs12" {
+			secretValue, err = getCertBundleForPKCS(*secretResp.Value, certBundleOnly, certKeyOnly)
+			if err != nil {
+				return nil, err
+			}
+		}
+		secretsMap[secretName] = []byte(secretValue)
+		if withTags {
+			appendTagsToSecretMap(secretName, secretsMap, secretResp.Tags)
+		}
+	} else {
+		secretList, err := basicClient.GetSecrets(context.Background(), "https://"+vaultName+".vault.azure.net", nil)
```

Comment:
```suggestion
		secretList, err := basicClient.GetSecrets(context.Background(), fmt.Sprintf("https://%s.vault.azure.net", vaultName), nil)
```

This may also help align with configurable Azure `vaultEndpoint`'s discussed above.